src/world/map.c

```diff
@@ -580,66 +581,64 @@ void map_invalidate_tile(int x, int y, int zLow, int zHigh)
  *
  *  rct2: 0x006E0E01
  */
-money32 map_try_clear_scenery(int x, int y, rct_map_element *mapElement, int flags)
+void game_command_remove_scenery(int* eax, int* ebx, int* ecx, int* edx, int* esi, int* edi, int* ebp)
 {
+	int x = *eax;
+	int y = *ecx;
+	uint8 base_height = *edx;
+	uint8 scenery_type = *edx >> 8;
+	uint8 map_element_type = *ebx >> 8;
 	money32 cost;
-	rct_scenery_entry *entry;
-
-	entry = g_smallSceneryEntries[mapElement->properties.scenery.type];
+	
+	rct_scenery_entry *entry = g_smallSceneryEntries[scenery_type];
 	cost = entry->small_scenery.removal_price * 10;
 
 	RCT2_GLOBAL(RCT2_ADDRESS_NEXT_EXPENDITURE_TYPE, uint8) = RCT_EXPENDITURE_TYPE_LANDSCAPING * 4;
-	RCT2_GLOBAL(0x009DEA5E, uint32) = x * 32 + 16;
-	RCT2_GLOBAL(0x009DEA60, uint32) = y * 32 + 16;
-	RCT2_GLOBAL(0x009DEA62, uint32) = mapElement->base_height * 8;
-
-	x *= 32;
-	y *= 32;
+	RCT2_GLOBAL(0x009DEA5E, uint32) = x + 16;
+	RCT2_GLOBAL(0x009DEA60, uint32) = y + 16;
+	RCT2_GLOBAL(0x009DEA62, uint32) = base_height * 8;
 
-	if (!(flags & 0x40) && RCT2_GLOBAL(RCT2_ADDRESS_GAME_PAUSED, uint8) != 0) {
+	if (!(*ebx & 0x40) && RCT2_GLOBAL(RCT2_ADDRESS_GAME_PAUSED, uint8) != 0) {
 		RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, uint16) = STR_CONSTRUCTION_NOT_POSSIBLE_WHILE_GAME_IS_PAUSED;
-		return MONEY32_UNDEFINED;
+		*ebx = MONEY32_UNDEFINED;
+		return;
 	}
 
-	if (!(RCT2_GLOBAL(RCT2_ADDRESS_SCREEN_FLAGS, uint8) & SCREEN_FLAGS_SCENARIO_EDITOR) && !(flags & 0x40)) {
+	if (!(RCT2_GLOBAL(RCT2_ADDRESS_SCREEN_FLAGS, uint8) & SCREEN_FLAGS_SCENARIO_EDITOR) && !(*ebx & 0x40)) {
 		// Check if allowed to remove item
 		if (RCT2_GLOBAL(RCT2_ADDRESS_PARK_FLAGS, uint32) & PARK_FLAGS_FORBID_TREE_REMOVAL) {
 			if (entry->small_scenery.height > 64) {
 				RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, uint16) = STR_FORBIDDEN_BY_THE_LOCAL_AUTHORITY;
-				return MONEY32_UNDEFINED;
+				*ebx = MONEY32_UNDEFINED;
+				return;
 			}
 		}
 
 		// Check if the land is owned
-		if (!(RCT2_GLOBAL(RCT2_ADDRESS_SCREEN_FLAGS, uint8) & SCREEN_FLAGS_SCENARIO_EDITOR))
-			if (!map_is_location_owned(x, y, RCT2_GLOBAL(0x009DEA62, uint32)))
-				return MONEY32_UNDEFINED;
+		if (!map_is_location_owned(x, y, RCT2_GLOBAL(0x009DEA62, uint32))){
+			*ebx = MONEY32_UNDEFINED;
+			return;
+		}
 	}
 
-	if ((flags & 0x40) && !(mapElement->flags & 0x10))
-		return 0;
+	rct_map_element* map_element = map_get_surface_element_at(x / 32, y / 32);
```

Comment:
This is not a surface element you are trying to find. So you should be using `map_get_first_element_at()` instead of get surface element. I'm not sure of the order of elements on a tile but its possible by using get_surface_element you will have gone past the element you are after.